client/components/data/query-products-list/index.jsx

```diff
@@ -1,10 +1,14 @@
 import { useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import { requestProductsList } from 'calypso/state/products-list/actions';
-import { isProductsListFetching } from 'calypso/state/products-list/selectors';
+import { isProductsListFetching, getProductsList } from 'calypso/state/products-list/selectors';
 
 const request = ( props ) => ( dispatch, getState ) => {
-	if ( isProductsListFetching( getState() ) ) {
+	const productsList = getProductsList( getState() );
+	if (
+		isProductsListFetching( getState() ) ||
+		( productsList && Object.keys( productsList ).length > 0 )
+	) {
```

Comment:
Instead of introducing such a big change for all components that rely on this (and might need fresh data), would it be safer if we added a prop to the component like `refetch: false/true` which by default would be `true` so that we keep backwards compatibility?

Reply:
That's a great point! 🎉  I've added a new prop called `persist`, so this way it's opt-in to use the store for caching the products list.